src/engine.cc

```diff
@@ -453,20 +456,42 @@ void EngineLoop::CmdPosition(const std::string& position,
                              const std::vector<std::string>& moves) {
   std::string fen = position;
   if (fen.empty()) {
-	  fen = ChessBoard::kStartposFen;
+    fen = ChessBoard::kStartposFen;
   }
   engine_.SetPosition(fen, moves);
 }
 
 void EngineLoop::CmdGetFen() {
- 	std::string fen = engine_.GetPosition();
-	return SendResponse(fen);
-
+  std::string fen = engine_.GetCurrentPositionFen();
+  return SendResponse(fen);
 }
 void EngineLoop::CmdGo(const GoParams& params) { engine_.Go(params); }
 
 void EngineLoop::CmdPonderHit() { engine_.PonderHit(); }
 
 void EngineLoop::CmdStop() { engine_.Stop(); }
 
+bool ContainsKey(const std::unordered_map<std::string, std::string>& params,
+	const std::string& key) {
+	return params.find(key) != params.end();
+}
+
+bool EngineLoop::CmdNonStandardUciCommand(const std::string & command, const std::unordered_map<std::string, std::string>& params)
+{
+	if (command == "get") {
```

Comment:
Why not just a `getfen`? Is it so in some other engine?
Also, maybe just pass unparsed command to `CmdNonStandardUciCommand` (not command and params), so that it can parse it as it wishes. That way we can even compare with `"get fen"` (because currently it would work with `get asd asd fen sdjksd sdfkls ldkjfds` but won't with `get asd fen sdjkfl sdjklsdjk`, which is not very nice).

Reply:
First I started out with 'getfen' but I did not like it as it is not English, and does not seem to match the other command/parameter pairs. I future we might want to support to get other stuff. 'get fen' is just a starting point. For analyses purposes we could extend further with other non-uci commands. Not sure what though, yet but for example 'get epd' if chess960 would be supported. It's not really part of the scope and when something new is introduced we can change this later.